### PR TITLE
Support Sign-In with Ethereum [EIP-4361] On Coordinape

### PR DESCRIPTION
**Pull request by FilipeAleixo**
This PR is meant to solve #375. Verified that post login functionality is intact after SIWE refactor.

The only thing missing is replacing this var with the actual domain https://github.com/EsteFilipe/coordinape/commit/26c4017e758d3ea537e26f2afc827d7114eac6e8#diff-01abf081aaf21c9dd7749fceb2eaf3c000f88207e51f4a1ab234455045e4d2deR27

### CODE REVIEW

**Comment by levity** [comment]
I'm in the middle of a big reworking atm; can someone create a short video or gif of the user-facing change?

**Comment by levity** [comment]
Or if you're satisfied, @crabsinger, that's sufficient

**Comment by EsteFilipe** [comment]
@crabsinger @levity pushed the allowed domains changes to #967 let me know what you think.

**Comment by crabsinger** [comment]
@EsteFilipe i tried to resolve a merge conflcit through the github UI, but it didn't go well. If you can force push from your original branch and then rebase your branch off main, this should get back to a better state.

**Comment by crabsinger** [comment]
> @EsteFilipe i tried to resolve a merge conflcit through the github UI, but it didn't go well. If you can force push from your original branch and then rebase your branch off main, this should get back to a better state.

ok, i fixed the rebase and have a new branch here: https://github.com/coordinape/coordinape/tree/feat/siwe

**Comment by EsteFilipe** [comment]
@crabsinger gotcha, if there's something else I can help with let me know

File: api/login.ts [hunk 1]
```diff
@@ -10,41 +11,62 @@ import {
...
-import {
-  parseInput,
-  verifyContractSignature,
-  verifySignature,
-} from '../api-lib/signature';
+import { parseInput } from '../api-lib/signature';
 
 Settings.defaultZone = 'utc';
 
+const allowedDomains = (process.env.SIWE_ALLOWED_DOMAINS)?.split(",").filter(item => item !== "") || ["localhost:3000"];
```
[the remark this answers is not in the document]
Comment (crabsinger) [reply api/login.ts:18]: @EsteFilipe it is seeming like subdomains are not working, so we need a way to specify if we want them to work or not...

**Comment by crabsinger** [comment]
Closing this - will move ahead in PR #991

**Pull request closed without merging**